doc (react-aria): Add documentation for lack of semantic heading element on `AccordionHeader` (#21726)
microsoft/fluentui: packages/react-accordion/Spec.md

| Keyboard | Home       | Moves Focus | Moves focus to the first panel heading                      |
| Keyboard | End        | Moves Focus | Moves focus to the last panel heading                       |

## Accessibility

Accessibility behavior is built into the spec as much as possible. This section addresses specific issues that don't fit well with the standard definition of the component.

### No heading level on `AccordionHeader` by default

As described on [WAI-ARIA Roles, States, and Properties](https://www.w3.org/TR/wai-aria-practices/#wai-aria-roles-states-and-properties) documentation for accordion:

> Each accordion header button is wrapped in an element with role heading that has a value set for aria-level that is appropriate for the information architecture of the page.

> Every `AccordionHeader` should have as it's root an element with `role="heading"` and a proper `aria-level` attribute (or be a semantic heading element: `h1`, `h2`, `h3`, `h4`, `h5` or `h6`). This behavior is not implemented by default on `AccordionHeader` as it's impossible to predict which heading level will be required by the user. Requiring manual addition of such ARIA requirement.

Reviewer: ```suggestion
Every `AccordionHeader` should have as its root a semantic heading element: `h1`, `h2`, `h3`, `h4`, `h5` or `h6`. Alternatively `role="heading"` and a proper `aria-level` attribute. This behavior is not implemented by default on `AccordionHeader` as it's impossible to predict which heading level will be required by the user. Requiring manual addition of such ARIA requirement.
```

Reviewer: let's recommend the best practice first :)

```tsx
{/* No heading level by default */}
<AccordionHeader>This is a header</AccordionHeader>
{/* Generated html */}

> <div> {/* ⚠️ This is wrong according to WAI-ARIA specification ⚠️*/}

Reviewer: I would not say that it is wrong, rather it would be an improvement to use headings
